enclave-manager/web/src/client/KurtosisClientContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { KurtosisClient } from "./KurtosisClient";
+import { Alert, AlertDescription, AlertIcon, AlertTitle, Flex, Heading, Spinner } from "@chakra-ui/react";
+import { assertDefined, isDefined, isStringTrue, stringifyError } from "../utils";
+import { AuthenticatedKurtosisClient } from "./AuthenticatedKurtosisClient";
+import { LocalKurtosisClient } from "./LocalKurtosisClient";
+
+type KurtosisClientContextState = {
+  client: KurtosisClient | null;
+};
+
+const KurtosisClientContext = createContext<KurtosisClientContextState>({ client: null });
+
+export const KurtosisClientProvider = ({ children }: PropsWithChildren) => {
+  const [client, setClient] = useState<KurtosisClient>();
+  const [jwtToken, setJwtToken] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
+    const receiveMessage = (event: MessageEvent) => {
+      const message = event.data.message;
+      switch (message) {
+        case "jwtToken":
+          const value = event.data.value;
+          if (isDefined(value)) {
+            setJwtToken(value);
+          }
+          break;
```

Comment:
This is a necessary evil that is tied to the loading the kurtosis client with authentication details. Maybe this should be part of the AuthenticatedKurtosisClient?

Reply:
I have also since added another message to be passed. this is the `package` information that the UI should load. That part is not related to authentication and will be used whether we authenticate in the cloud context or if we run it locally.

Reply:
I think it's better that this is in KurtosisClientProvider as it keeps responsibilities seperate.
* KurtosisClientProvider is responsible for `providing a KurtosisClient` - so is responsible for knowing about things like windows and browser events.
* The AuthenticatedKurtosisClient is responsible for `communicating with the cloud api` and to do so it requires a jwt token, which is provided to it in its constructor.

Reply:
Got it, that makes sense. I wonder though if we can split out the jwt token part from this code and catch and provide it from some other place. The generic KurtosisClient shouldn't know about jwt tokens imo.

Reply:
> The generic KurtosisClient shouldn't know about jwt tokens imo.
I agree - and it doesn't